### PR TITLE
Implementation of Probabilistic PCA

### PR DESCRIPTION
**Pull request by wildart**
Added basic Probabilistic PCA implementation with following methods:
- Maximum Likelihood
- EM
- Bayesian

### CODE REVIEW

File: src/ppca.jl [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+        d = size(W, 1)
+        (isempty(mean) || length(mean) == d) ||
+            throw(DimensionMismatch("Dimensions of weight matrix and mean are inconsistent."))
+        new{T}(mean, W, var)
+    end
+end
+
+## properties
+
+MultivariateStats.indim(M::PPCA) = size(M.W, 1)
```
Comment (ararslan) [inline src/ppca.jl]: You don't need to prefix with the module name to overload functions defined in the module.

File: src/ppca.jl [hunk 2]
```diff
@@ -178,7 +171,9 @@ function MultivariateStats.fit{T<:AbstractFloat}(::Type{PPCA}, X::DenseMatrix{T}
     d, n = size(X)
 
     # process mean
-    mv = MultivariateStats.preprocess_mean(X, mean)
+    mv = preprocess_mean(X, mean)
+    (isempty(mv) || length(mv) == d) ||
+            throw(DimensionMismatch("Dimensions of weight matrix and mean are inconsistent."))
```
Comment (ararslan) [inline src/ppca.jl]: Too much indentation here. I'd just turn this into an explicit `if` block; I think it'd read a little nicer.

File: src/ppca.jl [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    # process mean
+    mv = preprocess_mean(X, mean)
+    (isempty(mv) || length(mv) == d) ||
+            throw(DimensionMismatch("Dimensions of weight matrix and mean are inconsistent."))
+
+    if method == :ml
+        Z = centralize(X, mv)
+        M = ppcaml(Z, mv, maxoutdim=maxoutdim, tol=tol)
+    elseif method == :em
+        S = Base.covm(X, isempty(mv) ? 0 : mv, 2)
```
Comment (ararslan) [inline src/ppca.jl]: I guess Julia 0.4 doesn't like this signature for `covm`. I'd be fine dropping 0.4 support at this point though.

**Comment by wildart** [comment]
Added compatibility with 0.4 to pass travis tests. Ready to merge.

File: test/ppca.jl [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+M = PPCA(Float64[], W, σ²)
+
+@test indim(M) == 5
+@test outdim(M) == 3
+@test mean(M) == zeros(5)
+@test loadings(M) == W
+@test var(M) == σ²
+
+T = inv(W'*W .+ σ²*eye(3))*W'
+@test_approx_eq transform(M, X[:,1]) T * X[:,1]
```
Comment (ararslan) [inline test/ppca.jl]: `@test_approx_eq A B` is deprecated in 0.6 in favor of `@test A ≈ B`.

Comment (wildart) [inline test/ppca.jl]: I think it would be proper to drop an old version support and rewrite everything when 0.6 will be release. New commands would require Compat bump which would be better to do in a separate PR.

File: docs/source/ppca.rst [hunk 5]
```diff
@@ -0,0 +1,187 @@
...
+    Yte = transform(M, Xte)
+
+    # reconstruct testing observations (approximately)
+    Xr = reconstruct(M, Yte)
+
+
+Core Algorithms
+~~~~~~~~~~~~~~~~~
+
+Two algorithms are implemented in this package: ``ppcaml``, ``ppcaem``, and ``bayespca``.
```
Comment (ararslan) [inline docs/source/ppca.rst]: You say two algorithms but list three things

File: docs/source/ppca.rst [hunk 6]
```diff
@@ -0,0 +1,187 @@
...
+
+    :param n: The number of observations.
+
+    :return: The resultant PPCA model.
+
+    :note: This function accepts two keyword arguments: ``maxoutdim``, ``tol``,  and ``tot``.
+
+    **Additional notes:**
+
+    - Function uses the ``maxoutdim`` parameter as an upper boundary when it automatically determs the latent space dimensionality.
```
Comment (ararslan) [inline docs/source/ppca.rst]: "determs" -> "determines"?
